app/frontend/Calendar/DisplayEvent/Invitation.svelte

```diff
@@ -14,88 +14,119 @@
+            label={$t`Change to Maybe *=> Maybe attend the meeting`}
+            onClick={onTentative}
+            icon={MaybeIcon}
+            classes="maybe font-normal" />
+        </ButtonMenu>
+      {:else}
+        <Button
+          label={$t`Confirm *=> Confirm to attend the meeting`}
           onClick={onAccept}
           icon={AcceptIcon}
           classes="accept font-normal" />
-        <MenuItem
-          label={$t`Change to Maybe *=> Maybe attend the meeting`}
-          onClick={onTentative}
-          icon={MaybeIcon}
-          classes="maybe font-normal" />
-      </ButtonMenu>
-    </hbox>
-  {/if}
+        <Button
+          label={$t`Reject *=> Decline to attend the meeting`}
+          onClick={onDecline}
+          icon={DeclineIcon}
+          classes="decline secondary font-normal" />
+        <ButtonMenu buttonIcon={ChevronDownIcon}>
+          <MenuItem
+            label={$t`Maybe *=> Maybe attend the meeting`}
+            onClick={onTentative}
+            icon={MaybeIcon}
+            classes="maybe font-normal" />
+        </ButtonMenu>
+      {/if}
+    {:else if message.invitationMessage}
+      <Button label={$t`Update calendar`} disabled={updateDisabled} onClick={onUpdate} classes="font-normal" />
+    {/if}
+  </hbox>
 </vbox>
 
 <script lang="ts">
   import type { EMail } from "../../../logic/Mail/EMail";
   import { InvitationMessage, InvitationResponse, type InvitationResponseInMessage } from "../../../logic/Calendar/Invitation";
   import InvitationDisplay from "./InvitationDisplay.svelte";
+  import AccountDropDown from "../../Shared/AccountDropDown.svelte";
   import Button from "../../Shared/Button.svelte";
   import AcceptIcon from "lucide-svelte/icons/check-check";
   import DeclineIcon from "lucide-svelte/icons/x";
   import MaybeIcon from "lucide-svelte/icons/circle-help";
   import ChevronDownIcon from "lucide-svelte/icons/chevron-down";
-  import { t } from "../../../l10n/l10n";
+  import { gt, t } from "../../../l10n/l10n";
   import ButtonMenu from "../../Shared/Menu/ButtonMenu.svelte";
   import MenuItem from "../../Shared/Menu/MenuItem.svelte";
 
   export let message: EMail;
+  let calendars: Collection<Calendar>;
+  let selectedCalendar: Calendar | undefined;
+  let myResponse: InvitationResponse = InvitationResponse.NoResponseReceived;
+  let updateDisabled: boolean | string = false;
+  let incomingInvitation;
+
+  $: if (message.event) {
+    loadCalendars();
+  }
+
+  async function loadEvent() {
+    await message.loadEvent();
+    loadCalendars();
+  }
+  function loadCalendars() {
+    calendars = message.getUpdateCalendars();
+    selectedCalendar = calendars.first;
+    updateDisabled = calendars.length ? false : gt`This event is not in your calendar`;
+    if (selectedCalendar) {
+      selectCalendar();
+    }
+  }
+  function selectCalendar() {
+    incomingInvitation = selectedCalendar.getIncomingInvitationFor(message);
+    myResponse = incomingInvitation.myResponse;
```

Comment:
This is strange. 

The account selector is there to allow the user to select which calendar the meeting will be placed in, at the moment when he clicks Accept. Once the user selected a response, we show that state when the user clicks on the message again (quite possibily on another day).

Am I reading that right that it will clear the response display once the user changed the calendar in the calendar selector? That's confusing, because the user might think that it cleared his response at the organizer, but it did not.

Reply:
Can we move the event to the other calendar, using `event.moveToCalendar()`?
If that is much more difficult, then disable the calendar selector after response, yes.

Reply:
I can look into that.